server/pulp/plugins/util/publish_step.py

```diff
@@ -746,7 +746,7 @@ def process_main(self):
         # for items where http & https are published to a separate directory
 
         _logger.debug('Copying tree from %s to %s' % (self.source_dir, timestamp_master_dir))
-        shutil.copytree(self.source_dir, timestamp_master_dir, symlinks=True)
+        os.system("cp -a %s %s" % (self.source_dir, timestamp_master_dir))
```

Comment:
This should pass an additional option to `cp ` that will ensure symlinks are preserved. Other codepaths that haven't been tested could be relying on that.